### ISSUE TITLE
Splunk Operator: Support ephemeral Indexes for test data clusters

### ISSUE THREAD

**Issue by marcusschiesser** [splunk/splunk-operator#957]
### Please select the type of request

Enhancement

### Tell us more

**Describe the request**
For our staging environment, we have an indexer cluster deployed. We need a couple of indexes for that we don’t care about persistence as we are just storing generated test data in them.

As it’s a staging environment, we are setting it up automatically in our deployment pipeline using YAML descriptors. AFIK there are two options to create indexes for an indexer cluster declaratively via a YAML descriptor:
1. Creating an `indexes.conf` file, wrapping it in an app, store it in S3 and deploy it via the App Framework with `scope: cluster`
2. Creating a SmartStore index

> Note: I am excluding the `apps_location` feature as it seems to be deprecated and it doesn’t work well declaratively with YAML

Option 1 has the precondition to use the App Framework (which we are not using right now) and furthermore, we would have to set up another app in our deployment pipeline that is just containing one `indexes.conf` file. We’re going therefore for Option 2, but it’s also too much configuration for just a test system and it’s filling up remote storage.

**Expected behavior**
Similar to the `ephemeralStorage` option for `varVolumeStorageConfig`, add an `ephemeral` config to the `spec` of the `ClusterMaster` resource. Each ephemeral index to be used will be defined as child attributes of `ephemeral`. 

Example:
```yaml
apiVersion: enterprise.splunk.com/v3
kind: ClusterMaster
metadata:
  name: cm
spec:
  ephemeral:
    indexes:
      - name: my_testindex1
      - name: my_testindex2
```

**Splunk setup on K8S**
- Using Operator 1.1.0

**Reproduction/Testing steps**
- After applying the YAML above, indexes will be created on each peer using ephemeral storage.

**K8s environment**
- AWS EKS 1.21

**Comment by iaroslav-nakonechnikov** [comment]
but if you set up DB_PATH to ephemeral storage - everything will be created on ephemeral storage... Doesn't it help? 

i tried to relocate everything here: #953 and maybe this thing will help you as a 3rd solution?

**Comment by marcusschiesser** [comment]
@iaroslav-nakonechnikov You're talking about setting `SPLUNK_DB`? But how would the operator then generate the `indexes.conf` that I need? Isn't it only generating them if I set Smartstore?
Also, I forgot to mention that there is a third option, modifying `defaults.yml`, but that not's feasible as I already have to configure it for injecting secrets (see workaround for #657)

**Comment by iaroslav-nakonechnikov** [comment]
yes, this only says Splunk to use define path where to store all SPLUNK_DB data. indexes are part of DB, so all indexes you'll ask to create - will be created in SPLUNK_DB path. At least it should. 

and yes, in our env we don't set default indexes, we create them via apps.

**Comment by marcusschiesser** [comment]
Ok, so that's not what I am talking about. My problem is that no `indexes.conf` is created automatically

**Comment by vivekr-splunk** [comment]
Hi @marcusschiesser  can you test with App framework instead of smartstore configuration.

**Comment by marcusschiesser** [comment]
@vivekr-splunk I don't understand. This is a feature request and not a bug, so what should I re-test?

**Comment by michal-tatusko-splunk** [comment]
Thank you for your feedback @marcusschiesser. We considered your request. SOK is a tool for the Splunk lifecycle management, for tweaking any Splunk-related configurations we recommend using App Framework and we won't support direct configuration of Splunk functionalities via CRDs. We suggest upgrading the Operator to the latest release (2.7.1) and leveraging App Framework to cover your use cases. I will close this issue.
